src/context.cc

```diff
@@ -197,16 +190,41 @@ void Context::change_buffer(Buffer& buffer)
 void Context::forget_buffer(Buffer& buffer)
 {
     m_jump_list.forget_buffer(buffer);
-    if (m_last_buffer.get() == &buffer)
-        m_last_buffer = nullptr;
 
     if (&this->buffer() != &buffer)
         return;
 
     if (is_editing() && has_input_handler())
         input_handler().reset_normal_mode();
 
-    change_buffer(m_last_buffer ? *m_last_buffer : BufferManager::instance().get_first_buffer());
+    auto last_buffer = this->last_buffer();
+    change_buffer(last_buffer ? *last_buffer : BufferManager::instance().get_first_buffer());
+}
+
+Buffer* Context::last_buffer() const
+{
+    if (not m_jump_list.size())
+        return nullptr;
+
+    const auto next_buffer = std::find_if(
```

Comment:
I understand, I think the same way. But then I thought that `JumpList` *is* a vector, and the private data it uses (such as the `m_current` index) is only used by the extra methods that interact with the storage.

Since the intent behind this structure is not to override a container but only to add state to it, I think it's not that bad if it makes its use simpler.

@mawww just drop this commit if needed, I'm fine either way.

Reply:
Yeah, probably both variants are fine. A compromise would be to expose a conversion operator (or a getter) returning `const Vector<Buffer*> &`, then this can stay roughly the same, and we can't accidentally modify the jumplist.